### PR TITLE
Activate PR analysis for contributions outside of SonarSource

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: travis.sh [hunk 1]
```diff
@@ -58,24 +58,37 @@ CI)
...
-    # Do not deploy a SNAPSHOT version but the release version related to this build
-    set_maven_build_version $TRAVIS_BUILD_NUMBER
     export MAVEN_OPTS="-Xmx1G -Xms128m"
-    mvn deploy sonar:sonar -B -e -V \
-        -Pdeploy-sonarsource \
+
+    if [ -n "${GITHUB_TOKEN-}" ]; then
+      strongEcho "External pull request"
+      # external PR : no deployment to repox
+      SONAR_PROJECT_VERSION=`maven_expression "project.version"`
```
Comment (tomverin) [inline travis.sh]: why don't you put this before the IF so you have it once for both conditions

File: travis.sh [hunk 2]
```diff
@@ -58,24 +58,37 @@ CI)
...
+  elif [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
     strongEcho "Build and analyze pull request"
-    SONAR_PROJECT_VERSION=`maven_expression "project.version"`
-    # Do not deploy a SNAPSHOT version but the release version related to this build
-    set_maven_build_version $TRAVIS_BUILD_NUMBER
     export MAVEN_OPTS="-Xmx1G -Xms128m"
-    mvn deploy sonar:sonar -B -e -V \
-        -Pdeploy-sonarsource \
+
+    if [ -n "${GITHUB_TOKEN-}" ]; then
```
Comment (tomverin) [inline travis.sh]: @benzonico you inverted the cases, you are running external pull request code when the GITHUB_TOKEN is present

File: travis.sh (hunk 1 above)
Comment (benzonico) [inline travis.sh]: indeed.

File: travis.sh (hunk 2 above)
Comment (benzonico) [inline travis.sh]: ouch, fixed.

**Pull request merged**